comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -139,3 +96,53 @@ func (c *collector) extractSBOMWithTrivy(_ context.Context, storedImage *workloa
 
 	return nil
 }
+
+func (c *collector) startScanResultHandler(ctx context.Context, resultChan <-chan sbom.ScanResult) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case result := <-resultChan:
```

Comment:
here you should check that the channel is not close
```suggestion
		case result, ok := <-resultChan:
		  if !ok { return }
```